src/pytesmo/scaling.py

```diff
@@ -333,30 +352,104 @@ def cdf_match(src, ref,
         Maximum allowed value, output data is capped at this value
     nbins: int, optional
         Number of bins to use for estimation of the CDF
+    minobs : int
+        Minimum desired number of observations in a bin.
+    ** kwargs: dict
+        keywords to be passed onto the gen_cdf_match() function
 
     Returns
     -------
     CDF matched values: numpy.array
         dataset src with CDF as ref
     '''
-
     percentiles = np.linspace(0, 100, nbins)
+    
+    if not minobs is None:      
+        percentiles = utils.resize_percentiles(src, percentiles, minobs)
+    
     perc_src = np.array(np.percentile(src, percentiles))
-    perc_src = unique_percentiles_interpolate(perc_src,
+    perc_src = utils.unique_percentiles_interpolate(perc_src,
                                               percentiles=percentiles)
     perc_ref = np.array(np.percentile(ref, percentiles))
-    perc_ref = unique_percentiles_interpolate(perc_ref,
+    perc_ref = utils.unique_percentiles_interpolate(perc_ref,
                                               percentiles=percentiles)
 
-    return gen_cdf_match(src, perc_src, perc_ref,
+    return gen_cdf_match(src, perc_src, perc_ref, ref=ref,
                          min_val=min_val, max_val=max_val,
-                         k=5)
+                         k=5, **kwargs)
 
+def cdf_beta_match(src, ref,
```

Comment:
Could you add a test for this method? Maybe one test whether the rescaling with minobs works, and one that makes sure the maths work as expected.

Reply:
For the second one, something like this should work:

```
from scipy.stats import norm

def test_cdf_beta_match():
    n = 1000
    c = np.linspace(1/n, 1-1/n, n)
    x = norm.ppf(c)
    y = 2 * x + 2
    y_scaled = cdf_beta_match(y, x)
   np.testing.assert_almost_equal(y, y_scaled)
```

Reply:
The tests for all scaling methods are already included

Reply:
Under tests/test_scaling.py::test_scaling_method(method)

Reply:
all scaling methods are tested

Reply:
Ah, nice, that's fine, I didn't see this in the diff, sorry.